test/node-http-test.js

```diff
@@ -0,0 +1,185 @@
+    await node.open();
+    await nclient.open();
+    await nclient.call('watch chain');
+
+    const mnemonic = Mnemonic.fromPhrase(phrase);
+    const priv = HDPrivateKey.fromMnemonic(mnemonic);
+    const type = network.keyPrefix.coinType;
+    const key = priv.derive(44, true).derive(type, true).derive(0, true);
+    const xkey = key.derive(0).derive(0);
+
+    pubkey = xkey.publicKey;
+    privkey = xkey.privateKey;
+
+    cbAddress = Address.fromPubkey(pubkey).toString(network.type);
+
+    nclient.bind('tree commit', (root, entry, block) => {
+      assert.ok(root);
+      assert.ok(block);
+      assert.ok(entry);
+
+      socketData.push({root, entry, block});
+    });
+
+    node.mempool.on('tx', (tx) => {
+      mempoolData[tx.txid()] = true;
+    });
+  });
+
+  beforeEach(() => {
+    socketData = [];
+    mempoolData = {};
+  });
+
+  after(async () => {
+    await nclient.close();
+    await node.close();
+  });
+
+  describe('Websockets', function () {
+    describe('tree commit', () => {
+      it('should mine 1 tree interval', async () => {
+        await mineBlocks(treeInterval, cbAddress);
+
+        assert.equal(socketData.length, 1);
+      });
+
+      it('should send the correct tree root', async () => {
+        const name = await nclient.execute('grindname', [5]);
+        const rawName = Buffer.from(name, 'ascii');
+        const nameHash = rules.hashName(rawName);
+
+        const u32 = Buffer.alloc(4);
+        bio.writeU32(u32, 0, 0);
+
+        const output = new Output({
+          address: cbAddress,
+          value: 0,
+          covenant: {
+            type: types.OPEN,
+            items: [nameHash, u32, rawName]
+          }
+        });
+
+        const mtx = new MTX();
+        mtx.addOutput(output);
+
+        const coins = await nclient.getCoinsByAddresses([cbAddress]);
+        coins.sort((a, b) => a.height - b.height);
+        const coin = Coin.fromJSON(coins[0]);
+
+        // assert that the utxo has matured
+        assert.ok(node.chain.height > coin.height + network.coinbaseMaturity);
+        mtx.addCoin(coin);
+
+        const addr = Address.fromPubkey(pubkey);
+        const script = Script.fromPubkeyhash(addr.hash);
+
+        const sig = mtx.signature(0, script, coin.value, privkey);
+        mtx.inputs[0].witness = Witness.fromItems([sig, pubkey]);
+
+        const valid = mtx.verify();
+        assert.ok(valid);
+
+        const tx = mtx.toTX();
+        await node.sendTX(tx);
+
+        // wait for tx event in mempool
+        await common.forValue(mempoolData, tx.txid(), true);
+
+        // get the chain state before mining blocks
+        const pre = await nclient.getInfo();
+
+        const mempool = await nclient.getMempool();
+        assert.equal(mempool[0], mtx.txid());
+
+        await mineBlocks(treeInterval, cbAddress);
+        assert.equal(socketData.length, 1);
+
+        const {root, block, entry} = socketData[0];
+
+        // the root is committed to in the next block
```

Comment:
nit: this comment should be updated. We are asserting that a new root was committed and that the websocket event was fired.